packages/web-runtime/src/services/auth/authService.ts

```diff
@@ -0,0 +1,201 @@
+import { extractPublicLinkToken, isPublicLinkContext, isUserContext } from '../../router'
+
+export class AuthService {
+  private clientService: ClientService
+  private configurationManager: ConfigurationManager
+  private store: Store<any>
+  private router: VueRouter
+  private userManager: UserManager
+  private publicLinkManager: PublicLinkManager
+
+  public initialize(
+    configurationManager: ConfigurationManager,
+    clientService: ClientService,
+    store: Store<any>,
+    router: VueRouter
+  ): void {
+    this.configurationManager = configurationManager
+    this.clientService = clientService
+    this.store = store
+    this.router = router
+  }
+
+  /**
+   * Initialize publicLinkContext and userContext (whichever is available, respectively).
+   *
+   * FIXME: at the moment the order "publicLink first, user second" is important, because we trigger the `ready` hook of all applications
+   * as soon as any context is ready. This works well for user context pages, because they can't have a public link context at the same time.
+   * Public links on the other hand could have a logged in user as well, thus we need to make sure that the public link context is loaded first.
+   * For the moment this is fine. In the future we might want to wait triggering the `ready` hook of applications until all available contexts
+   * are loaded.
+   *
+   * @param to {Route}
+   */
+  public async initializeContext(to: Route) {
+    if (!this.publicLinkManager) {
+      this.publicLinkManager = new PublicLinkManager({
+        clientService: this.clientService,
+        configurationManager: this.configurationManager,
+        store: this.store
+      })
+    }
+
+    if (isPublicLinkContext(this.router, to)) {
+      const publicLinkToken = extractPublicLinkToken(to)
+      if (publicLinkToken) {
+        await this.publicLinkManager.updateContext(publicLinkToken)
+      }
+    }
+
+    if (!this.userManager) {
+      this.userManager = new UserManager({
+        clientService: this.clientService,
+        configurationManager: this.configurationManager,
+        store: this.store
+      })
+      this.userManager.events.addAccessTokenExpired(async (...args) => {
+        console.error('AccessToken Expired：', ...args)
+        await this.handleAuthError(this.router.currentRoute)
+      })
+      this.userManager.events.addAccessTokenExpiring((...args) => {
+        console.debug('AccessToken Expiring：', ...args)
+      })
+      this.userManager.events.addUserLoaded(async (user) => {
+        console.debug(
+          `New User Loaded. access_token： ${user.access_token}, refresh_token: ${user.refresh_token}`
+        )
+        try {
+          await this.userManager.updateContext(user.access_token)
+        } catch (e) {
+          console.error(e)
+          await this.handleAuthError(this.router.currentRoute)
+        }
+      })
+      this.userManager.events.addUserSignedIn(() => {
+        console.log('user signed in')
```

Comment:
`console.info`

Reply:
`UserSignedIn` and `UserSignedOut` are raised from the `SessionMonitor` in `oidc-client-ts`. As we don't use that, we never get this event and should just get rid of it.